accounts/serializers.py

```diff
@@ -0,0 +1,62 @@
+from rest_framework import serializers
+from .models import CustomUser, Role
+from rest_framework.authtoken.serializers import AuthTokenSerializer
+from rest_framework_simplejwt.tokens import RefreshToken
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+
+
+class CustomUserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CustomUser
+        fields = (
+            "user_id",
+            "username",
+            "email",
+            "first_name",
+            "last_name",
+            "phone_number",
+            "is_hero",
+            "is_profile_updated",
+            "country",
+            "city",
+            "address_line1",
+            "address_line2",
+            "role",
+            "password",
+        )
+
+        extra_kwargs = {"password": {"write_only": True}}
+
+
+class UserRoleSerializers(serializers.ModelSerializer):
+    class Meta:
+        model = Role
+        fields = "__all__"
+
+
+class UserLoginSerializer(AuthTokenSerializer):
+    username_or_email = None  # Custom field to accept username or email
+
+    def validate(self, attrs):
+        username_or_email = attrs.get("username")
+        password = attrs.get("password")
+
+        user = None
+        UserModel = get_user_model()
+        try:
+            user = UserModel.objects.get(
+                Q(username=username_or_email) | Q(email=username_or_email)
+            )
+        except UserModel.DoesNotExist:
+            pass
+
+        if user is not None and user.check_password(password):
+            if user.is_active:
+                refresh = RefreshToken.for_user(user)
+                attrs["access_token"] = str(refresh.access_token)
+                return {"access_token": attrs["access_token"]}
+            else:
+                raise serializers.ValidationError({"error": "User is not active."})
```

Comment:
@AbuuIbtisaam Here, check the way you're returning the validation error, Here you're returning more than one error at the same time, We need to return each error at a time. 

![image](https://github.com/SahanaTeam/sahaget-bn/assets/95165363/ec3f272d-b02d-4bdb-bfea-fbdad5dd6b81)

So, try to check it out! My Advice here is to return only one error at time even if more than one error is there.

Suggestion:

1. Before access backend you can first check the Validity of the username or password(Check in the validation.py in the signup fix branch, you'll find function that can help you to validate it before access the database.)
like check the length of the username, check whether the password follows the password convention.

Other thing is the way we're returning the error, my advice is we can use python dictionary to return it

For instance:
{
"error":"Error Description"
}